IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -47,71 +47,122 @@ - (instancetype)init
     return self;
 }
 
-- (void)storeCredential:(MSIDCredentialCacheItem *)credential forKey:(MSIDCacheKey *)key
+- (void)storeItem:(id<MSIDJsonSerializable>)item inBucket:(NSString *)bucket forKey:(MSIDCacheKey *)key
 {
     dispatch_barrier_async(self.queue, ^{
-        [self.cacheObjects setObject:credential forKey:key];
+        
+        NSMutableDictionary *items = nil;
+        if (bucket && ![self.cacheObjects objectForKey:bucket])
+        {
+            items = [NSMutableDictionary new];
+        }
+        else
+        {
+            items = [self.cacheObjects objectForKey:bucket];
+        }
+        
+        [items setObject:item forKey:key];
+        [self.cacheObjects setObject:items forKey:bucket];
     });
 }
 
-- (void)mergeStorageItem:(MSIDMacCredentialStorageItem *)storageItem
+- (void)mergeStorageItem:(MSIDMacCredentialStorageItem *)storageItem inBucket:(NSString *)bucket
```

Comment:
would love to see some unit tests especially on the merging logic

Reply:
Dean has already added a bunch of tests for account merging which goes through the same code path. Will add more later if thats not an issue.